rdt/transformers/categorical.py

```diff
@@ -425,8 +437,41 @@ class LabelEncoder(BaseTransformer):
     values_to_categories = None
     categories_to_values = None
 
-    def __init__(self, add_noise=False):
+    def __init__(self, add_noise=False, order_by=None):
         self.add_noise = add_noise
+        if order_by not in [None, 'alphabetical', 'numerical_value']:
+            raise Error(
+                "order_by must be one of the following values: None, 'numerical_value' or "
+                "'alphabetical'"
+            )
+
+        self.order_by = order_by
+
+    def _order_categories(self, unique_data):
+        if self.order_by == 'alphabetical':
+            if unique_data.dtype.type not in [np.str_, np.object_]:
+                raise Error("The data must be of type string if order_by is 'alphabetical'.")
+
+            nans = pd.isna(unique_data)
+            if nans.any():
+                nan_rep = unique_data[nans][0]
+                unique_data = np.sort(unique_data[~nans])
+                unique_data = np.append(unique_data, [nan_rep])
+
+            else:
+                unique_data = np.sort(unique_data)
+
+        elif self.order_by == 'numerical_value':
+            if np.issubdtype(unique_data.dtype.type, np.number):
+                unique_data = np.sort(unique_data)
+
+            else:
+                raise Error(
+                    'The data must be numerical or able to be casted as a float if order_by '
```

Comment:
The error message should be updated, since we don't support "caste-able" to float anymore.